service/frontend/workflow_handler.go

```diff
@@ -965,6 +973,18 @@ func (wh *WorkflowHandler) RespondWorkflowTaskCompleted(
 	}
 	namespaceId := namespace.ID(taskToken.GetNamespaceId())
 
+	if request.UseVersioning && len(request.WorkerVersionStamp.GetBuildId()) == 0 {
+		return nil, errUseVersioningWithoutBuildID
+	}
+
+	// Copy WorkerVersionStamp to BinaryChecksum if BinaryChecksum is missing (small
+	// optimization to save space in the request).
+	if request.WorkerVersionStamp != nil {
+		if len(request.WorkerVersionStamp.BuildId) > 0 && len(request.BinaryChecksum) == 0 {
+			request.BinaryChecksum = request.WorkerVersionStamp.BuildId
+		}
+	}
```

Comment:
Please extract a function for this

Reply:
You mean for this and the similar code in PollWorkflowTaskQueue? The types are all different, it can't be one function

Reply:
Oh oops not sure what I saw here